src/test/K6/src/tests/platform/storage/deleteinstances_v1.js

```diff
@@ -0,0 +1,82 @@
+/*
+    Test data required: username and password, deployed app that requires level 2 login (reference app: ttd/apps-test)
+    Command: docker-compose run k6 run /src/tests/platform/storage/deleteinstances_v1.js
+    -e env=*** -e org=*** -e username=*** -e userpwd=*** -e level2app=*** -e appnames=app1,app2 -e appsaccesskey=*** -e sblaccesskey=***
+*/
+
+import { check } from 'k6';
+import * as sbl from '../../../api/platform/storage/messageboxinstances.js';
+import * as setUpData from '../../../setup.js';
+import { addErrorCount } from '../../../errorcounter.js';
+
+const userName = __ENV.username;
+const userPassword = __ENV.userpwd;
+const appOwner = __ENV.org;
+const level2App = __ENV.level2app;
+const environment = __ENV.env.toLowerCase();
+let appNames = __ENV.appnames;
+
+export const options = {
+  thresholds: {
+    errors: ['count<1'],
+  },
+};
+
+//Function to setup data and return AltinnstudioRuntime Token and user details
+export function setup() {
+  var aspxauthCookie = setUpData.authenticateUser(userName, userPassword);
+  var altinnStudioRuntimeCookie = setUpData.getAltinnStudioRuntimeToken(aspxauthCookie);
+  var data = setUpData.getUserData(altinnStudioRuntimeCookie, appOwner, level2App);
+  data.RuntimeToken = altinnStudioRuntimeCookie;
+  setUpData.clearCookies();
+  return data;
+}
+
+//Hard delete instances under a party id based on supplied app names
+export default function (data) {
+  const runtimeToken = data['RuntimeToken'];
+
+  const partyIds = [];
+  partyIds.push(data['partyId']);
+  //add party id of org for non prod environments
+  if (environment != 'prod') partyIds.push(data['orgNumberPartyId']);
+
+  var res, success, instances;
+  var instancesCount = 0;
+
+  try {
+    appNames = appNames.split(',');
+    appNames.push(level2App);
+  } catch (error) {
+    appNames = [];
+    appNames.push(level2App);
+  }
+  partyIds.forEach((partyId) => {
+    do {
+      //Find active instances under the party id to be deleted.
+      var queryModel = {
+        instanceOwnerPartyIdList: [partyId],
+      };
+      res = sbl.searchSblInstancesPost(runtimeToken, queryModel);
+
+      //Filter instances based on appName
+      instances = sbl.filterInstancesByAppName(appNames, res.body);
```

Comment:
Could app names been made a part of the original query?

Reply:
The search endpoint in the messagebox doesn't allow for searching multiple instances at once. we would have to re-work the logic to use the other storage endpoint. As this is just moving existing code I'd rather we do that in a sperate issue.